### ISSUE TITLE
MME crash during handle_paging_expiry 

### ISSUE THREAD

**Issue by jpad-freedomfi** [magma/magma#4423]
### Your Environment

- **Version:** 1.3.2 with patched MME for Timezone
- **Affected Component:** Access Gateway
- **Affected Subcomponent:** MME
- **Deployment Environment:**  bare-metal (AGW)

### Describe the Issue

We are seeing MME (same version deployed at 4 sites) crash and restart at one site. This MME image is based on source ZIP file for 1.3.2 release with patch for Time and Timezone correction.

To my knowledge this site has 2 differences to other sites:

1. Baicells CPEs vs. Airspan CPEs (all on latest SW 2.4.7 for Baicells and 1.2.1 for Airspan)
2. Inactivity timer is at default value 4s whereas at other sites we have set to 600s. 

**To Reproduce**
1. No exact steps, I think have inactivity timer low and Baicells UE, during paging it will occur periodically. 

**Expected behavior**
Whatever the paging failure is should not cause the restart of MME, it should fail procedure and move on. 

**Additional context**
Example syslog attached mme crash with traceback. I have bigger syslog (>200MB) showing many instances of this crash in 24hrs at one site. 
`SUMMARY: AddressSanitizer: heap-use-after-free /home/vagrant/magma/lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c:2236 in mme_app_handle_paging_timer_expiry`

[example_syslog_paging_expiry_mme_restart.txt](https://github.com/magma/magma/files/5833912/example_syslog_paging_expiry_mme_restart.txt)

**Comment by jpad-freedomfi** [comment]
MME built from:
https://github.com/magma/magma/archive/v1.3.2.tar.gz 

with patch to` lte/gateway/c/oai/tasks/nas/emm/sap/emm_send.c` based on https://github.com/magma/magma/pull/3998/commits/e5638c75a79947f0ce5f23be30bf098810819691

**Comment by amarpad** [comment]
@ulaskozat is this https://github.com/magma/magma/commit/f1be5d670331501f70c90fd6b6981950d05edd6f

**Comment by jpad-freedomfi** [comment]
@ulaskozat and @amarpad I was hoping changing inactivity timer to a higher value would prevent the context release / service request activity and reduce the instances of this crash but that is not the case. We are still seeing this issue like every 5-10 minutes making the site un-useable. 

Any guidance on a patch I could apply to keep this exception from taking down the MME?

**Comment by ulaskozat** [comment]
@jpad-freedomfi please upgrade to 1.3.3. Paging timer expiry after UE context is removed was leading to crashes.

**Comment by ulaskozat** [comment]
Or you can apply this patch on top of what you have:
https://github.com/magma/magma/pull/4302

**Comment by jpad-freedomfi** [comment]
thanks @ulaskozat for the pointer. applying the patch. There is a scary release note on 1.3.3. about rebooting AGW HW to fix a known issue. Testing the patched MME now.

**Comment by jpad-freedomfi** [comment]
ok. Applied patched MME using patch from #4302 and we still reproduce the issue. trying #4318

**Comment by ulaskozat** [comment]
@jpad-freedomfi that scary note was not intended to be scary. Although all the services are restarting, somehow kernel holds onto the port indefinitely in one testing environment.

**Comment by ulaskozat** [comment]
#4318 and #4302 are the same things. We should have a troubleshooting for this. Another known issue about the timers is the race condition https://github.com/magma/magma/issues/4414. But the signature you shared in syslog looked the same as the issue fixed by #4302.

**Comment by ulaskozat** [comment]
@jpad-freedomfi how is the issue reproduced?

**Comment by jpad-freedomfi** [comment]
@ulaskozat I had a dislexia moment. I applied #4320 not #4302 on my first attempt. I then reversed the patch and applied #4318. That appears to have fixed the issues. 

I do not have exact steps to reproduce. This is in live network. I can share what I know but basically we see flapping UEs, looks like 15 UEs looping (attach, detach, attach, detach) with explicit detach messaging about 10s after attach. We have 2 eNBs in 1 TAC so paging would hit both eNBs sometimes after detach 

the whole cycle took about 2 minutes. with 15 UEs doing it, we saw the MME crash every 5-10 min. 

attached is a pcap with the 1 cycle of a single UE for timing and messaging reference. 
[soledad_flap_example.pcapng.zip](https://github.com/magma/magma/files/5845906/soledad_flap_example.pcapng.zip)
